fop/fom.c

```diff
@@ -335,11 +340,23 @@ M0_INTERNAL bool m0_fom_invariant(const struct m0_fom *fom)
 		_0C(fom->fo_service->rs_type == fom->fo_type->ft_rstype);
 }
 
+static void m0_fom_check_deadlock(const struct m0_fom *fom)
+{
+	if (!M0_IN(fom->fo_type->ft_id, (M0_CAS_PUT_FOP_OPCODE,
+					 M0_CAS_DEL_FOP_OPCODE)))
+	    return;
+#ifndef __KERNEL__
+	if (m0_cas_fom_in_deadlock(fom))
+		exit(1);
```

Comment:
Can you please print out some log messages to indicate that deadlock might be discovered and we have to exit now?
So, when this process exits, the HA will treat it as failure and will restart it again.

Reply:
yes will add log message, it is expected that HA will restart the process which got exited, so currently we are testing exit() behavior with HA on cluster, will update here once testing done.

Reply:
Please use `M0_ASSERT()` or `M0_IMPOSSIBLE()` to clearly indicate that it's an abnormal termination of the process.

Reply:
exit(1) was suggested by team,  not sure how HA behave with if is use M0_ASSERT